cpp/include/cudf/detail/nvtx/ranges.hpp

```diff
@@ -37,15 +37,26 @@ using thread_range = ::nvtx3::domain_thread_range<libcudf_domain>;
  * @brief Convenience macro for generating an NVTX range in the `libcudf` domain
  * from the lifetime of a function.
  *
- * Uses the name of the immediately enclosing function returned by `__func__` to
- * name the range.
+ * Uses the given name or the name of the immediately enclosing function returned
+ * by `__func__` to name the range.
  *
  * Example:
  * ```
  * void some_function(){
  *    CUDF_FUNC_RANGE();
+ *    CUDF_FUNC_RANGE("custom_name"); // Or use a custom name
```

Comment:
This macro should not exist. The _only_ reason for `CUDF_FUNC_RANGE` is for when you want to use the name of the function as the name of the range.

If you want to provide a custom name, then use a range object directly:
```
cudf::thread_range rng{"custom_name"};
```

Reply:
Ah, good to know.

I will change this PR for several minor cleanups and doc improvements then. https://github.com/rapidsai/cudf/issues/6476 should be addressed separately.

Reply:
Thanks @jrhemstad -- I had not come across `cudf::thread_range` before.

@PointKernel Can you update the [Developer Guide's NVTX Ranges section](https://github.com/rapidsai/cudf/blob/branch-24.04/cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md#nvtx-ranges) to describe exactly this:

- Use `CUDF_FUNC_RANGE` if you want to use the name of the function as the name of the NVTX range
- Use `cudf::thread_range rng{"custom_name"};` to provide a custom name for the current scope's NVTX range

Reply:
2c45d7a24c506e8e9364a91a2886d5b81aabc8d0 Done.

Reply:
Related discussion with Jake
https://github.com/rapidsai/cudf/pull/11864#issuecomment-1271514499